### PR TITLE
Add damage dice support to spells

### PR DESCRIPTION
**Pull request by CarlosFdez**
image
image
image

### CODE REVIEW

**Comment by CarlosFdez** [comment]
I removed sneak attack from the PR. If this requires a clean magical trickster implementation, then we will need to refactor sneak attack, and that'll require a migration. Aka performed separately.

File: src/module/actor/modifiers.ts [hunk 1]
```diff
@@ -642,8 +642,8 @@ class DiceModifierPF2e implements BaseRawModifier {
             : null;
 
         this.predicate = new PredicatePF2e(params.predicate ?? []);
-        this.enabled = this.predicate.test([]);
-        this.ignored = !this.enabled;
+        this.enabled = params.enabled ?? this.predicate.test([]);
+        this.ignored = params.ignored ?? !this.enabled;
```
Comment (stwlam) [inline src/module/actor/modifiers.ts:646]: How about we just remove the empty predicate test?

Comment (CarlosFdez) [reply src/module/actor/modifiers.ts:646]: we could replace it for testing if the predicate is empty instead. I think the intent is to say "if its not tested and it needs to be, default to disabled". Might break something if we don't have anything at all here.

Comment (stwlam) [reply src/module/actor/modifiers.ts:646]: They get tested in @system/damage/weapon, so it's covered there.

**Pull request merged**